### ISSUE TITLE
Recommended Way to Access Current User Info and Role on the Symfony UX Vue Component

### ISSUE THREAD

**Issue by vitamin-code** [symfony/ux#1006]
Hi everyone,

What's the recommended or the safest way to access the user id & role on the vue component. So far, I only know 2 options. 
The first option is to pass it on the component like this:
```
<div {{ vue_component('Your name is', { 'roles': currentUserlRole,  'userId': app.user.id }) }}></div>
```
Suppose that the `currentUserRole` contains `['ROLE_USER', 'ROLE_VIEWER']` and the `userId` contains the unique id of the user.
But I'm thinking, since it's passing on dom, tricky user can modify the user role there.

The second option is to create a Web API controller that fetches user id with roles and calls that method on vue `onMounted` then pass it to the store. But the little downside for this is, I need to put that on all the `onMounted` vue component.

The usage of this scenario is that, I need to enable or disable certain features on the vue component depending on the user and its role.

Please let me know if there's a better and safest way to achieve it without compromising security.

**Comment by weaverryan** [comment]
Hi there!

Passing information as the 2nd argument to `vue_component()` IS the correct way to do this, though I may not expose my literal role names, but that's up to you. Anyway, what you're effectively doing is informing your Vue component what the user "should" be able to do. But ultimately, no matter HOW you do this, the end result is that the server will need to check real security before doing anything.

For example, suppose you have a `ROLE_EDITOR` and you will show some extra buttons if the user has this. So, you pass this role to `vue_component()` like you are now, then render the buttons. There's no problem with this - if a bad user hacks in some extra roles, they just see some extra buttons. The key is that, if the user tries to use these buttons and, for example, attempts to "save" some changes to an article, when the Ajax request is sent to save, you will check if the user has permission to do this, an deny if they don't.

tl;dr (A) you are doing it correct, (B) no matter how you do this in Vue, a bad user can spoof it (C) but that's ok because before you perform any actions or return any sensitive data, you will check real security on the server.

Cheers!
